### PR TITLE
docs: Improve BGP Control Plane page

### LINKED ISSUE #23817: BGP Control Plane is not used
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

The cilium-agent tries to use MetalLBSpeaker even when I set `bgpControlPlane.enabled=true` in helm values. I'm trying to announce PodIPs in a IPv6 single-stack cluster. I have set the router-id on nodes.

### Cilium Version

1.13.0

### Kernel Version

6.1.10-200.fc37.x86_64

### Kubernetes Version

1.25.6

### Sysdump

_No response_

### Relevant log output

```shell
level=info msg="processing new event." component=MetalLBSpeaker.run subsys=bgp-speaker
level=info msg="Advertising CIDRs to all available session" advertisements="[<redacted>]" component=MetalLBSpeaker.announcePodCidrs subsys=bgp-speaker
level=error msg="Failed to announce pod CIDRs" CIDRs="&[<redacted>]" cidrs="&[<redacted>]" component=MetalLBSpeaker.handleNodeEvent error="cannot advertise non-v4 prefix \"<redacted>\"" labels="&map[beta.kubernetes.io/arch:amd64 beta.kubernetes.io/os:linux hailstone.cz/hypervisor:bare-metal kubernetes.io/arch:amd64 kubernetes.io/hostname:isengard kubernetes.io/os:linux node-role.kubernetes.io/control-plane: node.kubernetes.io/exclude-from-external-load-balancers:]" subsys=bgp-speaker
```

### Anything else?

Relevant helm values:
```
autoDirectNodeRoutes: true

bgp:
  enabled: true
  announce:
    loadBalancerIP: true
    podCIDR: true

bgpControlPlane:
  enabled: true

bpf:
  lbExternalClusterIP: true


ipam:
  mode: kubernetes

ipv4:
  enabled: false

ipv6:
  enabled: true

ipv6NativeRoutingCIDR: "{{ ipv6_lan_network }}"

enableIPv6Masquerade: false

kubeProxyReplacement: strict
kubeProxyReplacementHealtzBindAddr: "[::]:10256"

k8sServiceHost: "{{ ipv6_addr }}"
k8sServicePort: 6443

loadBalancer:
  mode: dsr

tunnel: disabled
```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by krouma**
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Thanks for contributing!

Fixes: #23817

Clarifies installation of BGP Control Plane
Fixes non-functional hyperlinks

### CODE REVIEW

File: Documentation/network/bgp-control-plane.rst [hunk 1]
```diff
@@ -19,6 +19,21 @@ Currently a single flag in the ``Cilium Agent`` exists to turn on the
 
    --enable-bgp-control-plane=true
 
+Following values have to be set for helm installation
```
Comment (qmonnet) [inline Documentation/network/bgp-control-plane.rst]: OK, I'd maybe rephrase like this:

```suggestion
If using Helm charts instead, the relevant values are the following:
```

Nit: You also have a typo in the commit description for this change (`arument` -> `argument`).

File: Documentation/network/bgp-control-plane.rst [hunk 2]
```diff
@@ -19,6 +19,21 @@ Currently a single flag in the ``Cilium Agent`` exists to turn on the
...
+   bgpControlPlane:
+     enabled: true
+
+.. note::
+
+   The BGP Control Plane feature is mutually exclusive with the :ref:`bgp`
+   feature and therefore it has to be disabled.
+   If both are enabled, the BGP will be prioritized and will probably fail.
+   In other words, this feature does not switch the BGP implementation
+   from MetalLB to GoBGP.
```
Comment (qmonnet) [inline Documentation/network/bgp-control-plane.rst]: This is a useful note, but is a bit confusing at times (it's hard to understand which BGP implementation we're talking about, and GoBGP/MetalLB aren't otherwise introduced in the page). Also if one has to be disabled anyway, I wouldn't try guessing what happens when both are enabled, let's just stick to “one has to be off”. Some suggestions below:

```suggestion
   The BGP Control Plane feature is mutually exclusive with the MetalLB-based :ref:`bgp`
   feature. To use the Control Plane, the older BGP feature has to be disabled.
   In other words, this feature does _not_ switch the BGP implementation
   from MetalLB to GoBGP.
```

**Review by qmonnet — CHANGES_REQUESTED** [review]

**Review by qmonnet — CHANGES_REQUESTED** [review]
Thank you, the changes look all good.

Could you please fold the new commits (4 and 5) into the relevant commits (1 and 2 I think)? You can drop the `Co-authored-by:` tags, in particular because GitHub generates them with my personal email instead of work email.

**Review by qmonnet — APPROVED** [review]
Looks good to me, thank you!

Re-adding review request to Dylan, which was presumably removed by accident.

**Comment by krouma** [comment]
Yes, I just clicked to rerequest review and didn't notice it removed the second one.

**Comment by qmonnet** [comment]
Dylan: Just thinking, while reviewing this PR again: If the two different BGP features are mutually exclusive, should we consider a follow-up PR to make the Cilium agent fail to start if both are required by the user?

**Comment by dylandreimerink** [comment]
To be honest, I don't know if the two are mutually exclusive, I will double check this for myself. However, I can recall a few issues where I have seem users enable both at the same time and them being confused as to why configuration from the legacy feature doesn't apply to the BGP Control Plane.

I think there fundamentally isn't an conflict since both use different methods of configuration and do not interfere on a network level as far as I know. And I would not enforce mutual exclusion in the agent since that would make migrating from MetalLB to BGP CP harder.

I still agree with the spirit of this doc change, but I suggest we change it be advisory and keep the clarification that enabling BGP CP does not disable the legacy BGP feature.

**Review by dylandreimerink — APPROVED** [review]
